---
title: F test theory
---


```{r}
suppressWarnings(suppressPackageStartupMessages({
	library(dplyr)
	library(knitr)
}))
knitr::opts_chunk$set(warning=FALSE, message=FALSE, cache=TRUE, echo=TRUE)
```

Simulate a simple 3 locus haplotype system with two loci in LD of 0.5, and each with MAF = 0.1

$$
\begin{aligned}
y_1 &\sim y_2 + y_3 + e_1 \\
y_1 &\sim y_2 + y_3 + y_{2}y_{3} + e_2
\end{aligned}
$$


Method to create y1, y2, y3:

```{r}
gendat <- function(n)
{
	y1 <- rep(0, n)
	y2 <- rep(0, n)
	y1[856:900] <- 1
	y2[901:945] <- 1
	y1[946:1000] <- 1
	y2[946:1000] <- 1
	y3 <- rbinom(n, 1, 0.5)
	return(tibble(y1,y2,y3))
}
```

Method to run the simulation:

```{r}
run_sim <- function(i, n=1000)
{
	set.seed(i)
	dat <- gendat(n)
	mod1 <- lm(y1 ~ y2 + y3, dat)
	mod2 <- lm(y1 ~ y2 + y3 + y2*y3, dat)
	amod <- anova(mod1, mod2)
	vare2 <- var(residuals(mod2))
	sse1 <- sum(residuals(mod1)^2)
	sse2 <- sum(residuals(mod2)^2)
	return(list(amod=amod, vare2=vare2, sse1=sse1, sse2=sse2))
}
```

Run

```{r}
l <- lapply(1:1000, function(x) run_sim(x))
```


Get the mean and variance of the F values


```{r}
sapply(l, function(x) x$amod$F[2]) %>% mean
sapply(l, function(x) x$amod$F[2]) %>% var
```

What about two trans effects? Expect the mean and variance of the F value to be 1 and 2, respectively.


```{r}
run_sim2 <- function(i, n=1000)
{
	set.seed(i)
	dat <- tibble(
		y1=rbinom(n, 1, 0.5),
		y2=rbinom(n, 1, 0.5),
		y3=rbinom(n, 1, 0.5)
	)
	mod1 <- lm(y1 ~ y2 + y3, dat)
	mod2 <- lm(y1 ~ y2 + y3 + y2*y3, dat)
	amod <- anova(mod1, mod2)
	vare2 <- var(residuals(mod2))
	sse1 <- sum(residuals(mod1)^2)
	sse2 <- sum(residuals(mod2)^2)
	return(list(amod=amod, vare2=vare2, sse1=sse1, sse2=sse2))
}
l2 <- lapply(1:1000, function(x) run_sim2(x))
sapply(l2, function(x) x$amod$F[2]) %>% mean
sapply(l2, function(x) x$amod$F[2]) %>% var
```


Peter wrote that 

$$
\begin{aligned}
F &= \frac{SSE_1 - SSE_2}{SSE_2 / (n-4)} \\
  &= \frac{Q}{var(e_2)} 
\end{aligned}
$$

where in the case of no interaction we expect

$$
\begin{aligned}
SSE_2 / (n-4) \approx var(e1) = var(e2) &= var(y_1)(1 - r^2) \\
&= p_1(1-p_1)(1 - r^2)
\end{aligned}
$$

and $Q = SSE_1 - SSE_2$. Test this:


```{r}
dat <- tibble(
	sim = 1:1000,
	F = sapply(l, function(x) x$amod$F[2]),
	vare2 = sapply(l, function(x) x$vare2),
	sse1 = sapply(l, function(x) x$sse1),
	sse2 = sapply(l, function(x) x$sse2)
)
dat2 <- tibble(
	sim = 1:1000,
	F = sapply(l2, function(x) x$amod$F[2]),
	vare2 = sapply(l2, function(x) x$vare2),
	sse1 = sapply(l2, function(x) x$sse1),
	sse2 = sapply(l2, function(x) x$sse2)
)
```

For the cis-trans association:

```{r}
plot(sse2 ~ sse1, dat)
```

For the trans-trans association:

```{r}
plot(sse2 ~ sse1, dat2)
```

Checking the equation

```{r}
plot((dat$sse1-dat$sse2)/dat$vare2, dat$F)
```

This works great. 

## Interaction test

Let the genotypic value of $y_{1.ij} = (y_1 = 1 | y_2 = i, y_3 = j)$ where $i,j \in \{0,1\}$ the counts

$$
n_{ij} = n ( 1- p_2 + i(2p_2) - 1))(1 - p_3 + i(2p_3-1))
$$

We can therefore calculate the numbers in the 2 x 2 combinations of $y_2$ and $y_3$:

$$
\begin{aligned}
y_2\; & y_3 & n_{ij}/n \\
0\; & 0 & (1 - p_2)(1 - p_3) \\
0\; & 1 & p_2(1 - p_3) \\
1\; & 0 & (1 - p_2)p_3 \\
1\; & 1 & p_2p_3 \\
\end{aligned}
$$

A test for interaction can be defined as

$$
\delta = mean(y_{1.11}) + mean(y_{1.00}) - mean(y_{1.10}) - mean(y_{1.01})
$$

with $mean(y_{1.ij}) = \sum y_{1.ij} / n_{ij}$. 

Check this. First generate a dataset with a strong interaction

```{r}
set.seed(sapply(l, function(x) x$amod$F[2]) %>% which.max)
dat <- gendat(1000)
mod1 <- lm(y1 ~ y2 + y3, dat)
mod2 <- lm(y1 ~ y2 + y3 + y2*y3, dat)
amod <- anova(mod1, mod2)

SSE1 <- sum(residuals(mod1)^2)
SSE2 <- sum(residuals(mod2)^2)
pchisq((SSE1 - SSE2) / (SSE2 / 996), 1, low=F)
pf((SSE1 - SSE2) / (SSE2 / 996), 1, 996, low=F)
```

Now create one with a weak interaction:

```{r}
set.seed(sapply(l, function(x) x$amod$F[2]) %>% which.min)
datw <- gendat(1000)
modw1 <- lm(y1 ~ y2 + y3, datw)
modw2 <- lm(y1 ~ y2 + y3 + y2*y3, datw)
amodw <- anova(modw1, modw2)
```

Test for interaction:

```{r}
est_delta <- function(dat)
{
	n <- nrow(dat)
	obs = table(dat$y1, dat$y2) / nrow(dat) %>% c	
	tab <- with(dat, tapply(y1, list(y2, y3), mean)) %>% c
	return(tibble(
		y1.00 = tab[1],
		y1.10 = tab[2],
		y1.01 = tab[3],
		y1.11 = tab[4],
		delta = y1.00 + y1.11 - y1.10 - y1.01,
		p1 = mean(dat$y1),
		p2 = mean(dat$y2),
		p3 = mean(dat$y3),
		p00 = obs[1],
		p10 = obs[2],
		p01 = obs[3],
		p11 = obs[4],
		D = obs[4] - mean(dat$y1) * mean(dat$y2)
	))
}

# strong interaction
est_delta(dat)

# weak interaction
est_delta(datw)
```

Variance of $\delta$ can be calculated from the allele and haplotype frequencies and compared to its variance under a linear model (LM). Create a set of simulations for cis-trans and trans-trans cases

```{r}
# cis-trans
run_sim3 <- function(i, n=1000)
{
	set.seed(i)
	dat <- gendat(n)
	return(est_delta(dat))
}
m <- lapply(1:1000, function(i) run_sim3(i)) %>% bind_rows()

# trans-trans
run_sim4 <- function(i, n=1000)
{
	set.seed(i)
	dat <- tibble(
		y1=rbinom(n, 1, 0.5),
		y2=rbinom(n, 1, 0.5),
		y3=rbinom(n, 1, 0.5)
	)
	return(est_delta(dat))
}
m2 <- lapply(1:1000, function(i) run_sim4(i)) %>% bind_rows()
```

From the given haplotype frequencies:

$$
\begin{aligned}
E(y_{1.11}) = E(y_{1.10}) &= p(y_1 = 1 | y_2 = 1) \\
                          &= p_{11} / p2 \\
                          &= p_1 + D / p_2
\end{aligned}
$$

Check:

```{r}
mean(m$y1.11)
mean(m$y1.10)
(m$p11/m$p2)[1]
(m$p1 + m$D/m$p2)[1]
```

and similarly

$$
\begin{aligned}
E(y_{1.00}) = E(y_{1.01}) &= p(y_1 = 0 | y_2 = 0) \\
                          &= p_{00} / p2 \\
                          &= p_1 - D/(1 - p_2)
\end{aligned}
$$


Check:

```{r}
mean(m$y1.00)
mean(m$y1.01)
(m$p1 - m$D/(1-m$p2))[1]
```

Each of the terms has a binomial variance

$$
var(mean(y_{1.ij})) = E(y_{1.ij})(1 - E(y1.ij)) / n_{ij}
$$

Check:

```{r}

# Variance of mean(y_1.11)
(sd(m$y1.11) / sqrt(nrow(m)))^2

# According to the formula
mean(m$y1.11) * (1 - mean(m$y1.11)) / mean((m$p2) * (m$p3) * 1000)
```

**Don't seem to be getting agreement here.**



